### PR TITLE
Migrate StaticDataProvider and BlobDataProvider to ZeroMap

### LINKED ISSUE #1056: Migrate BlobProvider to ZeroMap
We should be fully unblocked from migrating StaticDataProvider / BlobProvider from LiteMap to ZeroMap.

### PR DESCRIPTION
**Pull request by sffc**
Fixes #1056
Depends on #1057

Memory characteristics:

Before:

```
[memory] >      Running `target/release/examples/work_log`
[memory] > Feature
[memory] > dhat: Total:     20,642 bytes in 94 blocks
[memory] > dhat: At t-gmax: 20,274 bytes in 84 blocks
[memory] > dhat: At t-end:  1,024 bytes in 1 blocks
```

After:

```
[memory] >      Running `target/release/examples/work_log`
[memory] > Feature
[memory] > dhat: Total:     1,890 bytes in 93 blocks
[memory] > dhat: At t-gmax: 1,522 bytes in 83 blocks
[memory] > dhat: At t-end:  1,024 bytes in 1 blocks
```

### CODE REVIEW

**Comment by Manishearth** [comment]
> 2\. I cannot get buffers with a `'static` lifetime from a `ZeroMap<'static, String, Vec<u8>>`, unlike `LiteMap<&'static str, &'static [u8]>`; I can only get ephemeral references. This prevents me from using `SerdeDeDataReceiver::receive_static` and producing a 100% no-allocation data struct. The workaround is to allocate the `Rc`, but it's sad to do that, because I didn't have to do it before.

Honestly this makes me wonder if SDP _should_ use LiteMap. There's no extra allocation cost there that ZeroMap helps avoid, ZeroMap is only useful for _blob_ data providers. Perhaps SDP can be a different type with code shared with BlobDataProvider?

Or perhaps we can give up this particular property.

**Comment by sffc** [comment]
Actually what I was thinking what that this would be another use case for a "borrowed-only" version of ZV.

**Comment by sffc** [comment]
After a long while, this is ready for review!  Memory characteristics posted in the OP.

**Review by Manishearth — APPROVED** [review]

File: provider/blob/src/blob_data_provider.rs [hunk 1]
```diff
@@ -80,13 +86,10 @@ impl BlobDataProvider {
...
         let path = path_util::resource_path_to_string(&req.resource_path);
-        self.blob
+        self.data
             .try_project_cloned_with_capture::<&'static [u8], String, ()>(
                 path,
-                move |blob, path, _| {
-                    let BlobSchema::V001(blob) = blob;
-                    blob.resources.get(&*path).ok_or(()).map(|v| *v)
-                },
+                move |zm, path, _| zm.get(&*path).ok_or(()),
```
Comment (zbraniecki) [inline provider/blob/src/blob_data_provider.rs:92]: `question (n-b)`: do you need the move here? Do you need the deref-ref?

**Review by zbraniecki — DISMISSED** [review]
lgtm!

**Review by zbraniecki — DISMISSED** [review]
lgtm!

**Comment by sffc** [comment]
Whoops!  Looks like I need to update my dependencies...

```
error: failed to select a version for the requirement `zerovec = "^0.4"`
candidate versions found which didn't match: 0.5.0
location searched: /home/runner/work/icu4x/icu4x/utils/zerovec
required by package `icu_provider_blob v0.4.0 (/home/runner/work/icu4x/icu4x/provider/blob)`
Error: The process '/usr/share/rust/.cargo/bin/cargo' failed with exit code 101
```

**Review by Manishearth — APPROVED** [review]

**Pull request merged**